.github/workflows/generate-summary.yaml

```diff
@@ -187,11 +191,70 @@ jobs:
         run: |
           pip install pygithub requests
-          python ./scripts/download_artifacts.py -hash ${{ inputs.gcchash }} -repo patrick-rivos/gcc-postcommit-ci -token ${{ secrets.GITHUB_TOKEN }} -prefix "${{ inputs.prefix }}"
+          python ./scripts/download_artifacts.py -hash ${{ inputs.gcchash }} -repo patrick-rivos/gcc-postcommit-ci -token ${{ secrets.GITHUB_TOKEN }} -prefix "${{ inputs.prefix }}" -build-logs -build-logs-dir ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
           ls previous_logs
+          if [ -d "${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}" ]; then
+            ls ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
+          fi
+
+      - name: Unzip previous build log artifacts
+        id: unzip-previous-build-log-artifacts
+        uses: ./.github/actions/common/unzip-all-zips
+        with:
+          input-dir: ${{ steps.build-log-setup.outputs.previous_build_log_zips_path }}
+          output-dir: ${{ steps.build-log-setup.outputs.previous_build_logs_path }}
+          include-pattern: "*-stderr.log"
+
+      - name: Download current build log artifacts
+        id: download-current-build-log-artifacts
+        if: github.event_name != 'workflow_dispatch'
+        uses: actions/download-artifact@v4
+        with:
+          pattern: "*-build-log"
+          merge-multiple: true
+          path: ${{ steps.build-log-setup.outputs.current_build_log_zips_path }}
+
+      - name: Download current build log artifacts manual
+        id: download-current-build-log-artifacts-manual
+        if: (github.event_name == 'workflow_dispatch') && (inputs.build_log_current_run_id != '')
+        uses: actions/download-artifact@v4
+        with:
+          pattern: "*-build-log"
+          merge-multiple: true
+          path: ${{ steps.build-log-setup.outputs.current_build_log_zips_path }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          run-id: ${{ inputs.build_log_current_run_id }}
+
+      - name: Unzip current build log artifacts
+        id: unzip-current-build-log-artifacts
+        uses: ./.github/actions/common/unzip-all-zips
+        with:
+          input-dir: ${{ steps.build-log-setup.outputs.current_build_log_zips_path }}
+          output-dir: ${{ steps.build-log-setup.outputs.current_build_logs_path }}
+          include-pattern: "*-stderr.log"
+
+      - name: Parse New Build Warnings
+        run: |
+          python ./scripts/parse_new_build_warnings.py --old-dir ${{ steps.build-log-setup.outputs.previous_build_logs_path }}/build --new-dir ${{ steps.build-log-setup.outputs.current_build_logs_path }}/build --output ${{ steps.build-log-setup.outputs.new_build_warnings_path }}
+          if [ -f "${{ steps.build-log-setup.outputs.new_build_warnings_path }}" ]; then
+            cat ${{ steps.build-log-setup.outputs.new_build_warnings_path }}
+          else
+            echo "New build logs didn't exist"
+          fi
```

Comment:
I think we should also upload the output as an artifact for easier searching. Looking through the github actions output is flakey and also if we're going to put it into a comment, it'll help set up for that even if we'll need to trim it to fit the max character length for a comment

Reply:
For reusability, it seems to be better to upload it as an artifact. I'll upload the build log warning comparison log similarly as the testsuite log comparison.